### PR TITLE
Fixed #18: Plugin fails to load on QGIS 3.22

### LINKED ISSUE #18: Plugin fails to load on QGIS 3.22/master
```
Couldn't load plugin 'processing_saga_nextgen' due to an error when calling its classFactory() method 

ModuleNotFoundError: No module named 'processing.algs.saga.SagaNameDecorator' 
Traceback (most recent call last):
  File "C:\OSGeo4W/apps/qgis/./python\qgis\utils.py", line 423, in _startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "C:\Users/giovanni/AppData/Roaming/QGIS/QGIS3\profiles\training/python/plugins\processing_saga_nextgen\__init__.py", line 22, in classFactory
    from .saga_nextgen_plugin import SagaNextGenProviderPlugin
  File "C:\OSGeo4W/apps/qgis/./python\qgis\utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/giovanni/AppData/Roaming/QGIS/QGIS3\profiles\training/python/plugins\processing_saga_nextgen\saga_nextgen_plugin.py", line 21, in 
    from processing_saga_nextgen.processing.provider import SagaNextGenAlgorithmProvider
  File "C:\OSGeo4W/apps/qgis/./python\qgis\utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/giovanni/AppData/Roaming/QGIS/QGIS3\profiles\training/python/plugins\processing_saga_nextgen\processing\provider.py", line 35, in 
    from .SagaAlgorithm import SagaAlgorithm
  File "C:\OSGeo4W/apps/qgis/./python\qgis\utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/giovanni/AppData/Roaming/QGIS/QGIS3\profiles\training/python/plugins\processing_saga_nextgen\processing\SagaAlgorithm.py", line 55, in 
    from processing.algs.saga.SagaNameDecorator import decoratedAlgorithmName, decoratedGroupName
  File "C:\OSGeo4W/apps/qgis/./python\qgis\utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
ModuleNotFoundError: No module named 'processing.algs.saga.SagaNameDecorator'
```

### PR DESCRIPTION
**Pull request by SirRhynus**
See #18

### CODE REVIEW

**Comment by nyalldawson** [comment]
Thanks @SirRhynus !

I think we should actually just pull those files in to this plugin itself, so that we aren't risking future breakage like this if upstream changes occur. Are you able to do that here?

**Comment by SirRhynus** [comment]
Those files were already in the plugin, just not used there. I've updated them from the upstream though.

Are there also efforts going on to update the SagaProvider core plugin to 7.*?

File: processing_saga_nextgen/processing/SagaAlgorithm.py [hunk 1]
```diff
@@ -108,9 +108,17 @@ def icon(self):
     def svgIconPath(self):
         return QgsApplication.iconPath("providerSaga.svg")
 
+    def flags(self):
+        # TODO - maybe it's safe to background thread this?
+        f = super().flags() | QgsProcessingAlgorithm.FlagNoThreading
+        if self.known_issues:
+            f = f | QgsProcessingAlgorithm.FlagKnownIssues
+        return f
```
Comment (nyalldawson) [inline processing_saga_nextgen/processing/SagaAlgorithm.py:116]:
```suggestion
```

No need to resurrect this -- I pulled it out of this plugin deliberately to test/confirm that it's safe, and haven't had any reports of issues relating to this!

**Comment by SirRhynus** [comment]
I might need to comment again before you see that it's already been done.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**